Sources/MongoSwift/ConnectionString.swift

```diff
@@ -194,18 +194,32 @@ internal class ConnectionString {
     }
 
     internal func setCredential(from credential: Credential) throws {
-        guard mongoc_uri_set_username(self._uri, credential.username) else {
-            throw InvalidArgumentError(message: "Cannot set username to \(String(describing: credential.username))")
+        if let username = credential.username {
+            guard mongoc_uri_set_username(self._uri, username) else {
+                throw InvalidArgumentError(message: "Cannot set username to \(username).")
+            }
+        } else {
+            throw InvalidArgumentError(message: "Credential must specify username.")
```

Comment:
according to the spec, username is optional for MONGODB-X509 and MONGODB-AWS:
https://github.com/mongodb/specifications/blob/master/source/auth/auth.rst#mongocredential